orttraining/orttraining/training_ops/cpu/nn/dropout_op.cc

```diff
@@ -94,18 +94,28 @@ Status DropoutGrad<T1, T2>::Compute(OpKernelContext* context) const {
   auto mask_span = mask->DataAsSpan<bool>();
   const Tensor* ratio = context->Input<Tensor>(2);  // optional
   const float ratio_value = GetRatioOrDefault<T2>(ratio);
+  const Tensor* training_mode = context->Input<Tensor>(3); // optional
+
   const auto& dY_shape = dY->Shape();
   Tensor* dX = context->Output(0, dY_shape);
   auto dX_span = dX->MutableDataAsSpan<T1>();
+  bool training_mode_value = false;
+  if (training_mode) {
+    ORT_ENFORCE(training_mode->Shape().Size() == 1, "training_mode input should have a single boolean value.");
+    training_mode_value = *training_mode->Data<bool>();
+  }
 
   ORT_ENFORCE(mask->Shape() == dY_shape, "dY and mask should have the same shape");
   ORT_ENFORCE(dX->Shape() == dY_shape, "dY and dX should have the same shape");
 
-  if (ratio_value == 0.0f) {
+  if (ratio_value == 0.0f or !training_mode_value) {
+    std::cout << "CPU INFERENCE MODE (" << ratio_value << "/" << training_mode_value << ") " << std::endl;
```

Comment:
i assume the output is for debugging. if you want to print output in other cases, consider using the logging mechanism (include/onnxruntime/core/common/logging/macros.h)

Reply:
yup, just for debug purpose. The final version won't have any log